### PR TITLE
update(partner network): fix partner network page

### PR DESCRIPTION
**Pull request by manojava-gk**
## Description

entire partner network data structure based on the update from the bpdm server. move pageloading table js file to shared folder along with the helper file

## Why

page is broken

## Issue

NA

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

**Review by oyo — CHANGES_REQUESTED** [review]
Unit tests failing. Run `yarn test` to check. Also please remove commented out code

File: src/utils/dataMapper.ts [hunk 1]
```diff
@@ -18,67 +18,32 @@
...
-      )[0].value,
-      country: bpAddress.country.name,
-      street: bpAddress.thoroughfares[0].value,
-      zipCode: bpAddress.postCodes[0].value,
-      city: bpAddress.localities[0].value,
-      identifiers: bp.businessPartner.identifiers?.filter(
-        (identifier) => identifier.type.technicalKey !== 'CDQID'
-      ),
-    } as PartnerNetworkDataGrid
+      ...bp,
```
Comment (oyo) [inline src/utils/dataMapper.ts:36]: why do we need to clone these objects?

File: src/utils/dataMapper.ts [hunk 2]
```diff
@@ -18,67 +18,32 @@
...
-    legalForm: bp.legalForm?.name || '',
-    country: bpAddress.country.name,
-    street: bpAddress.thoroughfares[0].value,
-    zipCode: bpAddress.postCodes[0].value,
-    city: bpAddress.localities[0].value,
-    identifiers: bp.identifiers?.filter(
-      (identifier) => identifier.type.technicalKey !== 'CDQID'
-    ),
-  } as PartnerNetworkDataGrid
+    ...bp,
```
Comment (oyo) [inline src/utils/dataMapper.ts:45]: again why create a clone and not use the original objects?

**Review by oyo — CHANGES_REQUESTED** [review]
some new questions

File: src/features/partnerNetwork/types.ts [hunk 3]
```diff
@@ -121,30 +121,55 @@ export interface BpdmBusinessStatus {
...
-  content: Array<BusinessPartnerSearchResponse>
+  content: Array<BusinessPartner>
+}
+
+export interface BusinessPartnerAddressResponse {
+  alternativePostalAddress: any
+  bpnLegalEntity: string
+  createdAt: any
+  updatedAt: any
+  physicalPostalAddress: any
```
Comment (oyo) [inline src/features/partnerNetwork/types.ts:172]: define and use types or interfaces

File: src/features/partnerNetwork/types.ts [hunk 4]
```diff
@@ -162,12 +187,93 @@ export interface PartnerNetworkInitialState {
...
-  name: string
-  legalForm: string
-  country: string
-  street: string
-  zipCode: string
-  city: string
+  bpnl: string
+  legalName: string
+  legalForm: any
+  legalAddress: any
```
Comment (oyo) [inline src/features/partnerNetwork/types.ts:193]: define and use types or interfaces

File: src/features/partnerNetwork/types.ts [hunk 5]
```diff
@@ -121,30 +121,55 @@ export interface BpdmBusinessStatus {
...
-  addresses: Array<BpdmAddresses>
-  profileClassifications: Array<BpdmProfileClassification>
-  types: Array<BpdmTypeCommonKeyValuePair>
-  bankAccounts: Array<BpdmTypeBankAccount>
-  roles: Array<BpdmTypeCommonKeyValuePair>
+  states: [
+    {
+      description: string
+      validFrom: any
+      validTo: any
```
Comment (oyo) [inline src/features/partnerNetwork/types.ts:135]: define and use types or interfaces

File: src/features/newPartnerNetwork/types.ts [hunk 6]
```diff
@@ -1,52 +1,62 @@
...
-  legalAddress?: any
-  member?: boolean
-}
-
-export interface BusinessPartnerSearchResponse {
-  score: number
-  legalEntity: BusinessPartner
+  currentness: any
+  createdAt: any
+  updatedAt: any
```
Comment (oyo) [inline src/features/newPartnerNetwork/types.ts:60]: define and use types or interfaces

File: src/features/newPartnerNetwork/types.ts [hunk 7]
```diff
@@ -1,52 +1,62 @@
...
-  addresses: Array<BpdmAddresses>
-  profileClassifications: Array<BpdmProfileClassification>
-  types: Array<BpdmTypeCommonKeyValuePair>
-  bankAccounts: Array<BpdmTypeBankAccount>
-  roles: Array<BpdmTypeCommonKeyValuePair>
+  states: [
+    {
+      description: string
+      validFrom: any
+      validTo: any
```
Comment (oyo) [inline src/features/newPartnerNetwork/types.ts:40]: define and use types or interfaces

File: src/components/shared/basic/PageLoadingTable/index.tsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+  useEffect(() => {
+    if (loaded !== fetchHookRefresh) {
+      setLoaded(fetchHookRefresh)
+      setPage(0)
+      setClear(true)
+    }
+  }, [fetchHookRefresh, loaded])
+
+  /* eslint react-hooks/exhaustive-deps: "off" */
```
Comment (oyo) [inline src/components/shared/basic/PageLoadingTable/index.tsx:100]: no linter exceptions

File: src/components/shared/basic/PageLoadingTable/index.tsx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+  content: T[]
+}
+
+export interface PageLoadingTableProps extends Omit<TableProps, 'rows'> {
+  loadLabel: string
+  fetchHook: (paginArgs: PaginFetchArgs) => any
+  fetchHookArgs?: any
+  fetchHookRefresh?: number
+  allItems?: any
+  callbackToPage?: any
```
Comment (oyo) [inline src/components/shared/basic/PageLoadingTable/index.tsx:51]: define and use types or interfaces

File: src/components/shared/basic/PageLoadingTable/helper.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+export const hasMorePages = (data: any) => {
```
Comment (oyo) [inline src/components/shared/basic/PageLoadingTable/helper.ts:19]: define and use type or interface

File: src/components/shared/basic/PageLoadingTable/helper.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    data?.page < data?.totalPages - 1 ||
+    (data?.meta && data.meta.page < data.meta.totalPages - 1)
+  )
+}
+
+export const getMaxRows = (data: any) => {
```
Comment (oyo) [inline src/components/shared/basic/PageLoadingTable/helper.ts:26]: define and use type or interface

File: src/components/pages/PartnerNetwork/index.tsx [hunk 12]
```diff
@@ -132,10 +121,8 @@ const PartnerNetwork = () => {
           fetchHook={useFetchBusinessPartnersQuery}
           fetchHookArgs={{ expr }}
           fetchHookRefresh={refresh}
-          getRowId={(row: { legalEntity: any }) =>
-            row && row.legalEntity ? row.legalEntity.bpn : ''
-          }
-          columns={!showBPNColumn ? columns : bpnColumns}
+          getRowId={(row: any) => row.bpnl ?? ''}
```
Comment (oyo) [inline src/components/pages/PartnerNetwork/index.tsx:124]: define and use type or interface

File: src/components/shared/basic/PageLoadingTable/index.tsx [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+  Table,
+  TableProps,
+} from '@catena-x/portal-shared-components'
+import { Box } from '@mui/material'
+import { useState, useEffect } from 'react'
+import { getMaxRows, hasMorePages } from './helper'
+
+export interface PaginFetchArgs {
+  page: number
+  args?: any
```
Comment (oyo) [inline src/components/shared/basic/PageLoadingTable/index.tsx:30]: define and use type or interface

File: src/features/info/search/actions.ts [hunk 14]
```diff
@@ -178,9 +178,7 @@ const fetchSearch = createAsyncThunk(
               : item.name?.match(searchExpr) || item.provider.match(searchExpr)
           )
           .map((item: AppMarketplaceApp) => appToSearchItem(item)),
-        partners.content.map((item: { businessPartner: BusinessPartner }) =>
-          businessPartnerToSearchItem(item.businessPartner)
-        ),
+        partners.content.map((item: any) => businessPartnerToSearchItem(item)),
```
Comment (oyo) [inline src/features/info/search/actions.ts:181]: define and use types or interfaces

File: src/features/partnerNetwork/types.ts [hunk 15]
```diff
@@ -121,30 +121,55 @@ export interface BpdmBusinessStatus {
...
+  ]
   relations: Array<BpdmTypeRelation>
-}
-
-export interface BusinessPartnerSearchResponse {
-  score: number
-  businessPartner: BusinessPartner
+  currentness: any
+  createdAt: any
+  updatedAt: any
```
Comment (oyo) [inline src/features/partnerNetwork/types.ts:155]: define and use types or interfaces

**Review by oyo — CHANGES_REQUESTED** [review]
This code is way below our quality standards using too many any instead of proper types or interfaces. I think there is a fundamental issue in the coding philosophy: we should not try producing as much code as possible as quickly as possible. Instead we should try to produce good quality code.

Note that the comments are not covering all the code because I stopped reviewing after I saw  there are simply too many issues.

**Comment by oyo** [comment]
After the discussions and a closer look I think now the best approach is to transform the api response directly in the apiSlice using `transformResponse` or `handleResponse` adding a `meta` object with paging metadata. That way we can directly use the existing PageLoadingTable and don't have to duplicate a lot of code.

File: src/features/partnerNetwork/types.ts (hunk 4 above)
Comment (manojava-gk) [reply src/features/partnerNetwork/types.ts:199]: As discussed, this is needed for unit test case. but have removed any from all other places

File: src/features/info/search/actions.ts (hunk 14 above)
Comment (manojava-gk) [reply src/features/info/search/actions.ts:181]: As discussed, this is needed for unit test case.

**Review by oyo — CHANGES_REQUESTED** [review]
There is still the custom PageLoadingTable in this PR - remove and use the one from shared components after transforming the server response to same structure that the other APIs are using.

**Comment by manojava-gk** [comment]
> There is still the custom PageLoadingTable in this PR - remove and use the one from shared components after transforming the server response to same structure that the other APIs are using.

the one which is present in the shared component is the culprit. current file is the updated file. As part of clean up, I have moved generic JS file(as dropdown) back to portal. Let me know if this is not required

**Comment by oyo** [comment]
> > There is still the custom PageLoadingTable in this PR - remove and use the one from shared components after transforming the server response to same structure that the other APIs are using.
> 
> the one which is present in the shared component is the culprit. current file is the updated file. As part of clean up, I have moved generic JS file(as dropdown) back to portal. Let me know if this is not required

I mean this 
image
we should transform the API response to a structure with `meta` and `content` attributes just like all the other APIs. Then the PageLoadingTable from shared components should be able to handle the response and we don't need another one in this repo.

**Review by oyo — CHANGES_REQUESTED** [review]
The issue is that the partner network API uses a different structure for response than the other APIs. But it should be easy to transform into the common structure by moving the meta attributes about the paging information into a `meta` attribute. This way we can use the standard PageLoadingTable without modification and don't have to write a new one with custom code.

**Review by oyo — APPROVED** [review]
It's been decided to move the duplicate PageLoadingTable back to shared components in a followup PR.

**Pull request merged**